### DISCUSSION TITLE
Typo in Encode as String

### DISCUSSION THREAD

**Discussion by ArtichokeSap** [rsokl/Learning_Python#176] [Ideas]
Hey @rsokl , in the [Encode As String](https://www.pythonlikeyoumeanit.com/Module2_EssentialsOfPython/Problems/EncodeAsString.html) Problem in Module 2, there's a small typo, where it says:

> We use the str.join function along with a generator comprehensions in a couple places in our solution. Recall that
> 
> `"<hi>".join(x for x in some_iterable_of_strings)`
> 
> is equivalent to the long-form code:
> 
> ```
> out = ""
> for x in some_iterable_of_strings:
>     out += "<hi>" + x
> ```

But the former makes
`item1<hi>item2<hi>item3`
while the latter makes
`<hi>item1<hi>item2<hi>item3`

I would just make a pull request, but I'm not a natually pythonic coder to make equivalent, but still elegant code using a for loop; so I don't even know what to change it to.  For example, this does produce the output, but is ugly coding:
```
out = ""
num_elements = len(some_iterable_of_strings)
count = 0
for x in iterable_of_strings:
    count += 1
    if count < num_elements:
        out += x + '<hi>'
    else:
        out += x
```

Ultimately that doesn't satisfy me with the intention of "is equivalent to the long-form code".

**Comment by rsokl** [comment]
Thanks for catching this! I will try to cook up a good fix for this discrepancy. I appreciate you taking the time to post this.

**Reply by rsokl** [reply]
@ArtichokeSap I fixed this here: https://github.com/rsokl/Learning_Python/commit/96add6adca6f6d91c51ece24f943546cfd32d358

It will be reflected in the next HTML update (sorry for the *very* long delay!)
